index.js

```diff
@@ -63,6 +65,63 @@ async function fetchAllCustomers(defaultHeaders) {
     return customers
 }
 
+async function capturePaidInvoices(defaultHeaders, customerIgnoreRegex) {
+    let invoices = []
+
+    let paginationParam = ''
+    let invoiceJson = { has_more: true }
+
+    const today = new Date()
+    const firstDayThisMonth = new Date(today.getFullYear(), today.getMonth(), 1)
+    const firstDayNextMonth = new Date(today.getFullYear(), today.getMonth() + 1, 1)
+
+    const dateParam = `created[gte]=${parseInt((firstDayThisMonth.getTime() / 1000).toFixed(0))}&created[lt]=${parseInt(
+        (firstDayNextMonth / 1000).toFixed(0)
+    )}`
+    const statusParam = '&status=paid'
+
+    while (invoiceJson.has_more) {
+        const invoiceResponse = await fetchWithRetry(
+            `https://api.stripe.com/v1/invoices?limit=100&${dateParam}${statusParam}${paginationParam}`,
+            defaultHeaders
+        )
+        invoiceJson = await invoiceResponse.json()
+        const newPayments = invoiceJson.data
+
+        if (!newPayments) {
+            break
+        }
+
+        const lastObjectId = newPayments[newPayments.length - 1].id
+        paginationParam = `&starting_after=${lastObjectId}`
+        invoices = [...invoices, ...newPayments]
+    }
+
+    const cleanedPayments = []
+    invoices.forEach((payment) => {
+        if (!customerIgnoreRegex || !customerIgnoreRegex.test(payment.customer_email)) {
+            cleanedPayments.push({
```

Comment:
why do we need this? we don't seem to do anything with `cleanedPayments`? 

can't we just sum up the values in one loop?

Reply:
good catch – this was a remnant for another way i was thinking to handle these (log each payment / user). Removed!

Reply:
yeah I imagined that'd be the case